service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java

```diff
@@ -66,11 +67,16 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
             directionHelperService.collectDirectionsToMap(caseData);
 
         List<DirectionResponse> responses = directionHelperService.getResponses(directionsMap);
+        List<Element<Direction>> directionsToComplyWith = directionHelperService.getDirectionsToComplyWith(caseData);
 
-        directionHelperService.addResponsesToDirections(
-            responses, caseData.getStandardDirectionOrder().getDirections());
+        directionHelperService.addResponsesToDirections(responses, directionsToComplyWith);
 
-        caseDetails.getData().put("standardDirectionOrder", caseData.getStandardDirectionOrder());
+        //TODO: new service for sdo vs cmo in placing directions
+        if (caseData.getServedCaseManagementOrders().isEmpty()) {
+            caseDetails.getData().put("standardDirectionOrder", caseData.getStandardDirectionOrder());
+        } else {
+            caseDetails.getData().put("servedCaseManagementOrders", caseData.getServedCaseManagementOrders());
+        }
```

Comment:
Is it worth extracting this out even for testing sake?

Reply:
added the TODO and an integration test. I will move this at some point